docs/Migrating-Extensions-v2.0.md

```diff
@@ -0,0 +1,210 @@
+# Implementing "Extensions" in v2.0
+
+Parsedown v1.x allowed extensability through class extensions, where an developer
+could extend the core Parsedown class, and access or override any of the `protected`
+level methods and variables.
+
+Whilst this approach allows huge breadth to the type of functionality that can
+be added by an extension, it has some downsides too:
+
+* ### Composability: extensions cannot be combined easily
+  An extension must extend another extension for two extensions to work together.
+  This limits the usefulness of small extensions, because they cannot be combined with another small or popular extension.
+  If an extension author wishes the extension to be compatible with another extension, they can only pick one.
+
+* ### API stability
+  Because extensions have access to functions and variables at the `protected` API layer, it is hard to determine impacts of
+  internal changes. Yet, without being able to make a certain amount of internal change it is impractical to fix bugs or develop
+  new features. In the `1.x` branch, `1.8` was never released outside of a "beta" version for this reason: changes in the
+  `protected` API layer would break extensions.
+
+In order to address these concerns, "extensions" in Parsedown v2.0 will become more like "plugins", and with that comes a lot of
+flexability.
+
+ParsedownExtra is a popular extension for Parsedown, and this has been completely re-implemented for 2.0. In order to use
+ParsedownExtra with Parsedown, a user simply needs to write the following:
+
+```php
+$Parsedown = new Parsedown(new ParsedownExtra);
+$actualMarkup = $Parsedown->toHtml($markdown);
+```
+
+Here, ParsedownExtra is *composed* with Parsedown, but does not extend it.
+
+A key feature of *composability* is the ability to compose *multiple* extensions together, for example another
+extension, say, `ParsedownMath` could be composed with `ParsedownExtra` in a user-defined order.
+
+This time using the `::from` method, rather than the convinence constructor provided by `ParsedownExtra`.
+
+```php
+$Parsedown = new Parsedown(ParsedownExtra::from(ParsedownMath::from(new State)));
```

Comment:
Some user may gets confused by the `new State` as they there is no information in where this gets created in the first place.

Reply:
I'm tempted to leave this example as is: just because there are three other classes here that also should really be imported too 😉  The intent of this example is really to give examples how to combine various "extensions", but not really meant to be run without importing all the mentioned classes.